enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1158 @@
+   material change to the cluster.
+1. The control-plane update succeeds. The CVO, having reconciled its state, unsets metrics suggesting changes are pending.
+1. As part of updating cluster manifests, MachineConfigs have been modified. The MachineConfigOperator (MCO) re-renders a
+   configuration for worker-nodes. However, because the MCP maintenance schedule precludes initiating material changes,
+   it will not begin to update Machines with that desired configuration.
+1. The MCO will set a metric indicating that desired changes are pending. 
+1. `oc get -o=yaml/describe` will both provide status information indicating that changes are pending for the MCP and
+   the time at which the next material changes can be initiated according to the maintenance schedule.
+1. On the first Saturday of the next month, 00:00 UTC, the MCO determines that material changes are permitted.
+   Based on limits like maxUnavailable, the MCO begins to annotate nodes with the desiredConfiguration. The 
+   MachineConfigDaemon takes over from there, draining, and rebooting nodes into the updated release.
+1. There are a large number of nodes in the cluster and this process continues for more than 24 hours. On Saturday
+   23:59, the MCO applies a round of desired configurations annotations to Nodes. At 00:00 on Sunday, it detects
+   that material changes can no longer be initiated, and pauses its activity. Node updates that have already
+   been initiated continue beyond the maintenance schedule window.
+1. Since not all nodes have been updated, the MCO continues to expose a metric informing the system of
+   pending changes.
+1. In the subsequent days, the cluster is scaled up to handle additional workload. The new nodes receive
+   the most recent, desired configuration.    
+1. On the first Saturday of the next month, the MCO resumes its work. In order to ensure that forward progress is
+   made for all nodes, the MCO will update nodes that have the oldest current configuration first. This ensures
+   that even if the desired configuration has changed multiple times while maintenance was not permitted,
+   no nodes are starved of updates. Consider the alternative where (a) worker-node updates required > 24h,
+   (b) updates to nodes are performed alphabetically, and (c) MachineConfigs are frequently being changed
+   during times when maintenance is not permitted. This strategy could leave nodes sorting last 
+   lexicographically no opportunity to receive updates. This scenario would eventually leave those nodes
+   more prone to version skew issues.
+1. During this window of time, all node updates are initiated, and they complete successfully. 
+
+#### Service Delivery Emergency Patch
+1. SRE determines that a significant new CVE threatens the fleet.
+1. A new OpenShift release in each z-stream fixes the problem.
+1. SRE plans to override customer maintenance schedules in order to rapidly remediate the problem across the fleet.
```

Comment:
SRE have a couple of options and it depends on if it's HCP or not.

If HCP and the issue is signifigent enough, yes we would stage the update and force the change to land regardless of the customer maintenance schedule.

For Classic, where control plane is in customer's account, there is an important question to answer:  does this impact SRE's ability to provide support and/or impact SLA with customer?

If yes, there is a support and/or SLA impcat, then SRE would do the same as with HCP and pushed the upgrade through.

If no, the change would be staged and customer schedule will pick it up on the next update.  Customer will be notified and has an option to update earlier.  Customer can defer the update as usual.